### PR TITLE
nim: 0.20.2 -> 1.0.0

### PR DESCRIPTION
**Pull request by xzfc**
###### Motivation for this change

https://nim-lang.org/blog/2019/09/23/version-100-released.html

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @ehmry @peterhoeg

### CODE REVIEW

File: pkgs/development/compilers/nim/default.nix [hunk 1]
```diff
@@ -30,7 +30,7 @@ stdenv.mkDerivation rec {
   #    as part of building it, so it cannot be read-only
 
   checkInputs = [
-    nodejs-slim-11_x tzdata coreutils
+    nodejs-slim tzdata coreutils
```
Comment (xzfc) [inline pkgs/development/compilers/nim/default.nix]: Node 11.x is EOL (#69008), so I switched to current LTS.

**Comment by peterhoeg** [comment]
Looks good - thanks. Can I trouble you to remove me as a maintainer as part of this please?

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build nim

**Review by ghost — APPROVED** [review]
Looks fine, my only comment is that I think Nimble should be build seperately (99c24c5c71215559b0ae67bf411be408928cd7a0), but thats another issue ( #67878).

**Comment by Mic92** [comment]
It looks like two tests are failing on aarch64. Did this happen in the old version?

**Comment by xzfc** [comment]
> It looks like two tests are failing on aarch64. Did this happen in the old version?

No, these are new/updated tests.

----

PR update:
* Disabled broken tests on aarch64. I haven't tested though.
* Fortify hardening is disabled for tests and binaries created by nim compiler. It appends -O2 to gcc flags which is unwanted for unoptimized nim builds. Nim compiler itself is still built with fortify hardening enabled.
   * It works but I'm not sure that `wrapProgram --run` it a proper way of disabling it.
   * Re-enabled `tests/misc/tstrace.nim` that was broken by fortify hardening.

**Comment by xzfc** [comment]
@GrahamcOfBorg build nim

**Comment by xzfc** [comment]
Could anyone trigger `@GrahamcOfBorg build nim` please?

**Comment by Mic92** [comment]
@GrahamcOfBorg build nim

**Comment by marsam** [comment]
Thank you @xzfc

**Pull request merged**